framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1418,20 +1422,24 @@ protected void checkPreconditions(MethodInvocationTree tree, Set<Precondition> p
             }
 
             CFAbstractStore<?, ?> store = atypeFactory.getStoreBefore(tree);
-            CFAbstractValue<?> value = store.getValue(expr);
-
+            CFAbstractValue<?> value = null;
+            if (CFAbstractStore.canInsertReceiver(expr)) {
+                value = store.getValue(expr);
```

Comment:
Instead of guarding every call to `store.getValue`, we could change `CFAbstractStore#getValue(Receiver)` to return null if the receiver can't be inserted in the store.  Currently it [throws an exception](https://github.com/typetools/checker-framework/blob/master/framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java#L463).

Reply:
Wouldn't that mix "invalid key" and "no value for valid key"? Both would return null, but the behavior for them should be different. Explicitly handling and giving the user an error for "invalid key" seems better to me.

Reply:
> Explicitly handling and giving the user an error for "invalid key" seems better to me.

I was on the fence, so I'll leave the exception.